### PR TITLE
[cc26x2x7] Update build to match new OpenThread version

### LINKED ISSUE #7303: [TI] Update sample QR Codes in READMEs
#### Problem
While working on #7272 I spotted that sample QR Codes included in the TI example's READMEs have not been updated after recent changes in the QR Code format. E.g. the following QR code uses the old alphabet containing a space:
```
CH:.81TM -00 0C9SS0
```

 #### Proposed Solution
Update TI READMEs.

### PR DESCRIPTION
**Pull request by srickardti**
#### Problem

* Fixes build issues introduced by new OpenThread version
* Fixes #7303 
  * Removes specific language for QR codes within TI README files

#### Change overview

Fixes c pre-processor token difference with the current version of OpenThread.
Removes dependency on CLI library and UART code. Modifies documentation for
programming with CCS and UniFlash to point to public User's Guide.

#### Testing

* Manual build testing and run-to-main

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: src/platform/cc13x2_26x2/openthread-core-cc13x2_26x2-config.h [hunk 1]
```diff
@@ -52,7 +52,7 @@
  * Define to 1 to enable NCP UART support.
  *
  */
-#define OPENTHREAD_CONFIG_NCP_UART_ENABLE 0
+#define OPENTHREAD_CONFIG_NCP_HDLC_ENABLE 0
```
Comment (msandstedt) [inline src/platform/cc13x2_26x2/openthread-core-cc13x2_26x2-config.h:55]: Is this change intentional?  Comment no longer matches the #define.

Comment (srickardti) [reply src/platform/cc13x2_26x2/openthread-core-cc13x2_26x2-config.h:55]: Yes it was, the token was changed in OpenThread. It still is NCP over UART, but it uses the HDLC protocol over that UART.

I will also try building with the configuration removed.

Comment (msandstedt) [reply src/platform/cc13x2_26x2/openthread-core-cc13x2_26x2-config.h:55]: Sure, OK.  Can we change the comment though (`@def OPENTHREAD_CONFIG_NCP_UART_ENABLE`) to match?  The mismatch makes this look like a mistake.

**Review by msandstedt — APPROVED** [review]

File: src/platform/cc13x2_26x2/openthread-core-cc13x2_26x2-config.h (hunk 1 above)
Comment (msandstedt) [reply src/platform/cc13x2_26x2/openthread-core-cc13x2_26x2-config.h]: Thanks!  Cleaning unnecessary things out of the config is even better.
